src/odemis/gui/comp/viewport.py

```diff
@@ -1255,6 +1255,11 @@ def on_hrange(self, hrange):
             self.bottom_legend.lo_ellipsis = self.canvas.display_xrange[0] > self.canvas.data_xrange[0]
             self.bottom_legend.hi_ellipsis = self.canvas.display_xrange[1] < self.canvas.data_xrange[1]
 
+        if self.canvas.display_xrange != self.canvas.data_xrange:
+            self.hrange_lock.value = True
+        else:
+            self.hrange_lock.value = False
```

Comment:
Please add a comment to explain the goal of this code. Something like "Disable auto-scale as soon as the user manually zooms in".

More importantly, I don't think this does the right thing. It now also automatically re-enable auto-scale when zooming out. I don't think we should do that. Just let the user unlock it explicitly. 
(to be done both here, and in the vrange)